### PR TITLE
fix: wrong scope in captureMethod

### PR DESCRIPTION
**Pull request by ratscrew**
using the captureMethod decorator the "this" is no longer the original obj so changing to the target fixes this

## Description of your changes

change the binding of the decorated method back to the target so the below code will work:

```
export class Lambda implements LambdaInterface {
   @tracer.captureMethod()
   myMethod(){

      //other method is undefined as "this" is no longer the Lambda class
      this.otherMethod()
   }

   otherMethod(){}
}
```

### Related issues, RFCs

#1028 

### PR status

***Is this ready for review?:*** YES  
***Is it a breaking change?:*** NO

## Checklist

- [x] [My changes meet the tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [x] I have made corresponding changes to the *documentation*
- [x] I have made corresponding changes to the *examples*
- [x] My changes generate *no new warnings*
- [x] The *code coverage* hasn't decreased
- [x] I have *added tests* that prove my change is effective and works
- [ ] New and existing *unit tests pass* locally and in Github Actions
- [x] Any *dependent changes have been merged and published* in downstream module
- [x] The PR title follows the [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

- [ ] I have documented the migration process
- [ ] I have added, implemented necessary warnings (if it can live side by side)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by dreamorosi** [comment]
Hi @ratscrew thank you for taking the time to open a PR to this repo.

Before investing time in reviewing the PR, could you please open an issue to report the bug and provide also reproduction steps / example of this bug? We'd like to understand more about what's happening.

Then, after we have all agreed that it's in fact a bug, please review & edit the PR body and make sure to fill/check all the relevant sections.

**Comment by ratscrew** [comment]
@dreamorosi thanks for your quick response i have added in the bug issue.  

thanks

**Comment by dreamorosi** [comment]
Thank you for opening the issue and providing more info, as I mentioned in the issue I was able to reproduce the bug.

The fix that you are proposing seems like it would fix the issue. I was wondering if you could also please add a couple of tests to make sure it's the case and also to avoid future regressions.

For the unit tests you can use this one as a reference: https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/packages/tracer/tests/unit/Tracer.test.ts#L892-L934 and add one in that same file & test group. Ideally we should have a passing test similar to the one in your issue.

File: packages/tracer/src/Tracer.ts
Comment (dreamorosi) [inline packages/tracer/src/Tracer.ts:421]: This is mostly a styling comment, but could you please revert all the changes to this file except the one that contains the fix? (here I'm referring to the tabs/spaces, extra parenthesis, extra spaces, extra lines.

It's a bit difficult to see the actual diff & these would not be in line with the rest of the codebase.

**Review by dreamorosi — CHANGES_REQUESTED** [review]

**Comment by dreamorosi** [comment]
Thank you @ratscrew, we are really close, just some minor styling comments in one file - thanks for adding the unit test.

Looking forward to merge this!

File: packages/tracer/src/Tracer.ts
Comment (ratscrew) [reply packages/tracer/src/Tracer.ts]: Sorry did not see your comment at first, i have fixed this and re-committed

**Comment by dreamorosi** [comment]
It seems that the linting checks are not passing:

Screenshot 2022-07-28 at 18 24 28

According to the style we are using methods inside classes must be ordered alphabetically by name.

**Review by dreamorosi — APPROVED** [review]
Thank you @ratscrew for the contribution and for taking the time to address the review comments, looks good to me.

I'm going to request the review of a second maintainer as we have a 2-approvals model before merging.

Note to the next maintainer: the checks are failing due to #991 but **the tests, lint, etc. are OK**

**Comment by dreamorosi** [comment]
image
image
image


Ran integration tests manually, they are all passing

**Review by saragerion — APPROVED** [review]

**Pull request merged**
